File: node-app/v3_nativo.js
const express = require('express');
const app = express();
const PORT = 8000;

// Función nativa en JavaScript para convertir números a palabras en español
function numeroALetras(num) {
    if (num === 0) return "cero";

    const unidades = ["", "uno", "dos", "tres", "cuatro", "cinco", "seis", "siete", "ocho", "nueve"];
    const decenas = ["", "diez", "veinte", "treinta", "cuarenta", "cincuenta", "sesenta", "setenta", "ochenta", "noventa"];
    const especiales = ["diez", "once", "doce", "trece", "catorce", "quince", "dieciséis", "diecisiete", "dieciocho", "diecinueve"];
    const decenasMod = ["", "veinti", "treinta y ", "cuarenta y ", "cincuenta y ", "sesenta y ", "setenta y ", "ochenta y ", "noventa y "];

    if (num < 10) return unidades[num];
    if (num >= 10 && num < 20) return especiales[num - 10];
    
    if (num >= 20 && num < 100) {
        const d = Math.floor(num / 10);
        const u = num % 10;
        if (u === 0) {
            return d === 2 ? "veinte" : decenas[d];
        }
        return decenasMod[d - 1] + unidades[u];
    }
    
    if (num === 100) return "cien";
    
    // Soporte básico para números mayores por si pruebas con valores más altos
    return "número fuera de rango de prueba básica (0-100)";
}

// Ruta requerida por tu caso de estudio: http://localhost:8000/conintl?n=10
app.get('/conintl', (req, res) => {
    // Captura el parámetro 'n', si no existe usa "0" y lo fuerza a número entero
    const numero = parseInt(req.query.n || "0", 10);

    if (isNaN(numero)) {
        return res.status(400).send("Por favor, introduce un número válido.");
    }

    // Ejecuta la función local
    const resultadoPalabras = numeroALetras(numero);
    
    // Envía el resultado al navegador
    res.send(resultadoPalabras);
});

app.listen(PORT, () => {
    console.log(`Servidor nativo local corriendo en http://localhost:${PORT}`);
});

File: java-app/src/main/java/com/multilenguaje/v3_nativo.java
package com.multilenguaje;

public class v3_nativo {
    public static String ejecutar(String numeroStr) {
        int num = 0;
        try {
            num = Integer.parseInt(numeroStr != null ? numeroStr : "0");
        } catch (NumberFormatException e) {
            return "Número inválido";
        }

        if (num == 0) return "cero";

        String[] unidades = {"", "uno", "dos", "tres", "cuatro", "cinco", "seis", "siete", "ocho", "nueve"};
        String[] decenas = {"", "diez", "veinte", "treinta", "cuarenta", "cincuenta", "sesenta", "setenta", "ochenta", "noventa"};
        
        if (num < 10) return unidades[num];
        
        if (num == 11) return "once";
        if (num == 12) return "doce";
        if (num == 13) return "trece";
        if (num == 14) return "catorce";
        if (num == 15) return "quince";
        if (num == 16) return "dieciséis";
        if (num == 17) return "diecisiete";
        if (num == 18) return "dieciocho";
        if (num == 19) return "diecinueve";
        
        if (num == 20) return "veinte";
        if (num > 20 && num < 30) return "veinti" + unidades[num % 10];

        if (num < 100) {
            int u = num % 10;
            int d = num / 10;
            if (u == 0) return decenas[d];
            return decenas[d] + " y " + unidades[u];
        }
        
        if (num == 100) return "cien";

        return "Número fuera de rango (0-100)";
    }
}